eth/core/precompile/method_helpers.go

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2023, Berachain Foundation. All rights reserved.
+// Use of this software is govered by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package precompile
+
+import (
+	"reflect"
+	"unicode"
+
+	"pkg.berachain.dev/polaris/eth/accounts/abi"
+	errorslib "pkg.berachain.dev/polaris/lib/errors"
+)
+
+// this function finds the ABI method that matches the given impl method. It returns the key in the
+// ABI methods map that matches the impl method.
+func findMatchingABIMethod(
+	implMethod reflect.Method, precompileABI map[string]abi.Method,
+) (string, error) {
+	implMethodName := formatName(implMethod.Name)
+
+	for i := len(implMethodName) - 1; i >= 1; i-- {
```

Comment:
is this gonna end us up with some n^3 horribleness?

Reply:
it's ``O(m*n^2)`` where `m` is the length of the string name and `n` is the # of methods. `m` is usually very small since overloading is rare and `n` is the number of methods. we can assume # of inputs and # of outputs to be constant since they're small.